pinot-common/src/main/java/org/apache/pinot/common/metrics/ValidationMetrics.java

```diff
@@ -241,7 +242,7 @@ public void cleanupSegmentCountGauge(final String resource) {
 
   @VisibleForTesting
   public static String makeGaugeName(final String resource, final String gaugeName) {
-    return "pinot.controller." + resource + "." + gaugeName;
+    return "pinot.controller." + StringUtils.replaceOnce(resource, ".", "#") + "." + gaugeName;
```

Comment:
Could you please help me understand:
1. Why do we need to replace the first `.`?  Is it because the first part before the first `.` is the database name? Is this assumption true?
2. If we need to make the change for gauge, how about meter and timer?

Reply:
Make sense, will add java doc